Tests/MongoSwiftTests/DocumentTests.swift

```diff
@@ -680,4 +682,122 @@ final class DocumentTests: MongoSwiftTestCase {
         expect(doc["autoclosure test", default: "\(stringVal) and \(floatVal)" + stringVal])
             .to(bsonEqual("\(stringVal) and \(floatVal)" + stringVal))
     }
+
+    func testUUIDCodingStrategies() throws {
+        // TODO: add encoding strategy tests SWIFT-282
+
+        // swiftlint:disable nesting
+        struct UUIDWrapper: Codable {
+            let uuid: UUID
+        }
+        // swiftlint:enable nesting
+
+        // randomly generated uuid
+        guard let uuid = UUID(uuidString: "2c380a6c-7bc5-48cb-84a2-b26777a72276") else {
+            throw MongoError.bsonDecodeError(message: "Cant create UUID.")
+        }
+
+        let decoder = BSONDecoder()
+
+        // UUID default decoder expects a string
+        decoder.uuidDecodingStrategy = .deferredToUUID
+        let stringDoc: Document = ["uuid": uuid.description]
+        let badString: Document = ["uuid": "hello"]
+        let deferredStruct = try decoder.decode(UUIDWrapper.self, from: stringDoc)
+        expect(deferredStruct.uuid).to(equal(uuid))
+        expect(try decoder.decode(UUIDWrapper.self, from: badString)).to(throwError())
+
+        decoder.uuidDecodingStrategy = .fromString
+        let stringStruct = try decoder.decode(UUIDWrapper.self, from: stringDoc)
```

Comment:
is there a difference in functionality between `deferredToUUID` and `fromString`? they seem to work with the same data here.

Reply:
One explicitly invokes the `init(uuidString:)` and one invokes `init(from: decoder)` (which expects a string). So they do the same thing in different ways.

Reply:
is there a good reason to have both then?

Reply:
Actually, no, there isn't. I included both because I mistakenly thought `JSONDecoder` similarly duplicated the default in `Date`, but `.deferredToDate` expects seconds since _reference time_ instead of seconds from epoch/1970 which `.secondsSince1970` expects. I assumed they were duplicated for flexibility in case `init(from:)` ever changed, but it turns out there's no duplication at all.

I'll remove the `.fromString` strategy.